pkg/tempo/promsdprocessor/prom_sd_processor.go

```diff
@@ -0,0 +1,216 @@
+}
+
+// Start is invoked during service startup.
+func (p *promServiceDiscoProcessor) Start(_ context.Context, _ component.Host) error {
+	go p.watchServiceDiscovery()
+
+	go func() {
+		err := p.discoveryMgr.Run()
+		if err != nil {
+			level.Error(p.logger).Log("msg", "failed to start prom svc disco.  relabeling disabled", "err", err)
+		}
+	}()
+
+	return nil
+}
+
+// Shutdown is invoked during service shutdown.
+func (p *promServiceDiscoProcessor) Shutdown(context.Context) error {
+	return nil
+}
+
+func (p *promServiceDiscoProcessor) watchServiceDiscovery() {
+	for targetGoups := range p.discoveryMgr.SyncCh() {
+		hostLabels := make(map[string]model.LabelSet)
+		level.Debug(p.logger).Log("msg", "syncing target groups", "count", len(targetGoups))
+		for jobName, groups := range targetGoups {
+			p.syncGroups(jobName, groups, hostLabels)
+		}
+		p.mtx.Lock()
+		p.hostLabels = hostLabels
+		p.mtx.Unlock()
+	}
+}
+
+func (p *promServiceDiscoProcessor) syncGroups(jobName string, groups []*targetgroup.Group, hostLabels map[string]model.LabelSet) {
+	level.Debug(p.logger).Log("msg", "syncing target group", "jobName", jobName)
+	for _, g := range groups {
+		p.syncTargets(jobName, g, hostLabels)
+	}
+}
+
+func (p *promServiceDiscoProcessor) syncTargets(jobName string, group *targetgroup.Group, hostLabels map[string]model.LabelSet) {
+	level.Debug(p.logger).Log("msg", "syncing targets", "count", len(group.Targets))
+
+	relabelConfig := p.relabelConfigs[jobName]
+	if relabelConfig == nil {
+		level.Warn(p.logger).Log("msg", "relabel config not found for job. skipping labeling", "jobName", jobName)
+		return
+	}
+
+	for _, t := range group.Targets {
+		discoveredLabels := group.Labels.Merge(t)
+
+		level.Debug(p.logger).Log("discoveredLabels", discoveredLabels)
+		var labelMap = make(map[string]string)
+		for k, v := range discoveredLabels.Clone() {
+			labelMap[string(k)] = string(v)
+		}
+		processedLabels := relabel.Process(labels.FromMap(labelMap), relabelConfig...)
+		level.Debug(p.logger).Log("processedLabels", processedLabels)
+
+		var labels = make(model.LabelSet)
+		for k, v := range processedLabels.Map() {
+			labels[model.LabelName(k)] = model.LabelValue(v)
+		}
+
+		address, ok := labels[addressLabel]
+		if !ok {
+			level.Warn(p.logger).Log("msg", "ignoring target, unable to find address", "labels", labels.String())
+			continue
+		}
+
+		host := string(address)
+		if strings.Contains(host, ":") {
+			var err error
+			host, _, err = net.SplitHostPort(host)
+			if err != nil {
+				level.Warn(p.logger).Log("msg", "unable to split host port", "address", address, "err", err)
+				continue
+			}
+		}
+
+		for k := range labels {
+			if strings.HasPrefix(string(k), "__") {
+				delete(labels, k)
+			}
+		}
+
+		level.Debug(p.logger).Log("msg", "adding host to hostLabels", "host", host)
+		hostLabels[host] = labels
+	}
+}
+
+func (p *promServiceDiscoProcessor) processAttributes(attrs pdata.AttributeMap) {
+	// find the ip
```

Comment:
`ip` takes precedence over `net.host.ip`, right? Should the order be flipped here with a comment explaining the first tag name will be used? 

Is there ever a chance both tag names might exist but only one is the "correct" one?

Reply:
This is a hard question.  To some extent I'm coding to our environment which only includes the `ip` tag.  I lifted the otel tag from their docs, but i've never actually seen it.

Reply:
👍 fine with me